### ISSUE TITLE
Proc.new / &block warnings in recent install.

### ISSUE THREAD

**Issue by wolandscat** [asciidoctor/asciidoctor-bibtex#70]
I just upgraded Ubuntu 19 to 20.04, and asciidoctor-bibtex stopped working. So I re-installed it. This should show what got installed:
```
thomas:openEHR-specifications$ sudo gem install asciidoctor-bibtex
[sudo] password for thomas: 
Fetching asciidoctor-bibtex-0.7.1.gem
Fetching namae-1.0.1.gem
Fetching citeproc-1.0.10.gem
Fetching latex-decode-0.3.1.gem
Fetching bibtex-ruby-4.4.7.gem
Fetching csl-1.5.1.gem
Fetching citeproc-ruby-1.1.12.gem
Fetching csl-styles-1.0.1.10.gem
Successfully installed latex-decode-0.3.1
Successfully installed bibtex-ruby-4.4.7
Successfully installed namae-1.0.1
Successfully installed citeproc-1.0.10
Successfully installed csl-1.5.1
Successfully installed citeproc-ruby-1.1.12
Successfully installed csl-styles-1.0.1.10
Successfully installed asciidoctor-bibtex-0.7.1
Parsing documentation for latex-decode-0.3.1
Installing ri documentation for latex-decode-0.3.1
Parsing documentation for bibtex-ruby-4.4.7
Installing ri documentation for bibtex-ruby-4.4.7
Parsing documentation for namae-1.0.1
Installing ri documentation for namae-1.0.1
Parsing documentation for citeproc-1.0.10
Installing ri documentation for citeproc-1.0.10
Parsing documentation for csl-1.5.1
Installing ri documentation for csl-1.5.1
Parsing documentation for citeproc-ruby-1.1.12
Installing ri documentation for citeproc-ruby-1.1.12
Parsing documentation for csl-styles-1.0.1.10
Installing ri documentation for csl-styles-1.0.1.10
Parsing documentation for asciidoctor-bibtex-0.7.1
Installing ri documentation for asciidoctor-bibtex-0.7.1
Done installing documentation for latex-decode, bibtex-ruby, namae, citeproc, csl, citeproc-ruby, csl-styles, asciidoctor-bibtex after 3 seconds
8 gems installed
```
Here's the asciidoctor version info:
```
thomas:openEHR-specifications$ asciidoctor --version
Asciidoctor 2.0.10 [https://asciidoctor.org]
Runtime Environment (ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]) (lc:UTF-8 fs:UTF-8 in:UTF-8 ex:UTF-8)
```

Then when running it (from asciidcotor, in the usual way), I get the following warnings (voluminous):

```
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/bibliography.rb:150: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/entry.rb:162: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/entry.rb:162: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/entry.rb:162: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/entry.rb:162: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/entry.rb:162: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
```
This does not appear to have affected anything in the output.

**Comment by mojavelinux** [comment]
Ruby has been starting to clamp down on deprecations in recent versions, and these warnings are a reflection of those changes. They are just corelib warnings, so they're not going to affect the operation of the program (at least for now).

From the log message, I see that these warnings are not originating from this library. They're coming from bibtex-ruby. So it's best to file the issue with that project. https://github.com/inukshuk/bibtex-ruby

**Comment by mojavelinux** [comment]
This is the offending line: https://github.com/inukshuk/bibtex-ruby/blob/4.4.7/lib/bibtex/bibliography.rb#L150

It appears to be fixed in the latest release: https://github.com/inukshuk/bibtex-ruby/blob/5.1.4/lib/bibtex/bibliography.rb#L149

Thus, upgrading bibtex-ruby would be the right change for this library (revising my original statement).

**Comment by wolandscat** [comment]
Hm. I installed that, and now I have two versions:
```
thomas:openEHR-specifications$ gem info bibtex-ruby

*** LOCAL GEMS ***

bibtex-ruby (5.1.4, 4.4.7)
    Author: Sylvester Keil
    Homepage: http://inukshuk.github.com/bibtex-ruby
    License: GPL-3.0
    Installed at (5.1.4): /var/lib/gems/2.7.0
                 (4.4.7): /var/lib/gems/2.7.0

    A BibTeX parser, converter and API for Ruby.
```

But when I run my command, I still get this:

```
/var/lib/gems/2.7.0/gems/bibtex-ruby-4.4.7/lib/bibtex/entry.rb:162: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
```
which as you can see is still using the 4.4.7 version. I don't know how Ruby works, but I'm guessing that asciidoctor-bibtex needs to be upgraded (i.e. at source) to include the more recent bibtext-ruby? Or is there another way I can make it just use the most recent - although in general, that's probably something it is not tested with... so might be dangerous?

**Comment by mojavelinux** [comment]
Yes, the upgrade has to be in this library because that's what is
requesting it.

**Comment by mojavelinux** [comment]
It looks like it has already been upgraded in master:

https://github.com/asciidoctor/asciidoctor-bibtex/blob/master/asciidoctor-bibtex.gemspec#L19

So it's just a new release of this gem that is needed.

**Comment by wolandscat** [comment]
Ok great. Sorry for being a Ruby newbie, but only so many hours in the day!

**Comment by ProgramFan** [comment]
I am just fixing one nasty bug and a new release will be made after that in this week. If you can not wait, just clone this repo, uses `gem build asciidoctor-bibtex.gemspec` to generate a gem and install it with `gem install`.

**Comment by bwklein** [comment]
@ProgramFan have you released this fix yet?

**Comment by ProgramFan** [comment]
Fixed in 0.8.0.
